oak/server/channel.h

```diff
@@ -102,12 +103,24 @@ class MessageChannel {
 
   mutable absl::Mutex mu_;  // protects msgs_
   std::deque<std::unique_ptr<Message>> msgs_ GUARDED_BY(mu_);
+
+  mutable absl::Mutex count_mu_;  // protects read_count_, write_count_
+  int read_count_ GUARDED_BY(count_mu_);
```

Comment:
Maybe `reader_count_` and `writer_count_`? Otherwise it sounds like it's counting number of calls to `read`. Either way, a short comment here would be useful to clarify.